src/init.cpp

```diff
@@ -906,6 +906,10 @@ bool AppInitParameterInteraction(const ArgsManager& args)
         InitWarning(_("Option '-checkpoints' is set but checkpoints were removed. This option has no effect."));
     }
 
+    if (args.IsArgSet("-datacarriersize") || args.IsArgSet("-datacarrier")) {
+        InitWarning(_("Options '-datacarrier' or '-datacarriersize' are set but are marked as deprecated. They will be removed in a future version."));
```

Comment:
f4ae3126ea9650c4da9aa80c0f8c5d12da84be2f: in the GUI this results in a popup every time you start the node.

`m_warnings` could be used instead to treat it similar to the "This is a pre-release test build" message. Or we'd need a way to permanently dismiss a given warning string, which would be a GUI followup.

For similar discussion, see https://github.com/bitcoin/bitcoin/pull/31916#issuecomment-2685149599